portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import java.util.*; 
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.nio.charset.StandardCharsets;
+
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+  /**
+   * Delete all the comments from datastore  
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity comment: results.asIterable()) {
+      Key commentKey = comment.getKey();
+      datastore.delete(commentKey);
+    }
```

Comment:
Of course, in the real world you'd need authentication (i.e. "who are you?") & authorisation (i.e. "are you allowed to do this?") here! :)

Usually anything that creates, updates, or deletes any data should always have some sort of auth with it.  Sometimes you'll want auth for reading data too (e.g. looking at your bank account data), but often it is ok to read data without auth first (e.g. looking at a public instagram post or something)

Don't worry about it in this PR, but please do think about auth in your capstone project :)

Reply:
Thank you, I'll add authentication as a TODO for this week.